### PR TITLE
Improve XR pipeline detection to prevent regression

### PR DESCRIPTION
**Pull request by MaxWang-MS**
## Overview
Improve the XR pipeline detection logic and change the name of the property to better align with its behavior.

## Changes
- Fixes: #9817

### CODE REVIEW

**Review by davidkline-ms — APPROVED** [review]

File: Assets/MRTK/Core/MRTK.Core.asmdef [hunk 1]
```diff
@@ -2,7 +2,8 @@
     "name": "Microsoft.MixedReality.Toolkit",
     "references": [
         "Microsoft.MixedReality.Toolkit.Async",
-        "Microsoft.MixedReality.Toolkit.Editor.Utilities"
+        "Microsoft.MixedReality.Toolkit.Editor.Utilities",
+        "Unity.XR.Management"
```
Comment (keveleigh) [inline Assets/MRTK/Core/MRTK.Core.asmdef:6]: Not a huge fan of adding a dependency like this to the core, but I'm not sure another way around it with the APIs Unity provides...

Comment (davidkline-ms) [reply Assets/MRTK/Core/MRTK.Core.asmdef:6]: same here. we DO have similar entries in some of our other packages

Comment (keveleigh) [reply Assets/MRTK/Core/MRTK.Core.asmdef:6]: Some make sense to have the references (like providers that are built specifically on XR SDK), but the core is a bit different imo

Comment (MaxWang-MS) [reply Assets/MRTK/Core/MRTK.Core.asmdef:6]: We will be saying goodbye to this soon in 3.0 and also it feels more like a Unity-created issue so I'd vote for including that dependency. After all XR is pretty core to what we do, it's just not as core to Unity.

**Pull request merged**